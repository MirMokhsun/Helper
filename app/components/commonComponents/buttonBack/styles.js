import { StyleSheet, I18nManager } from 'react-native';
import Utils from '../../../api/utils/utils';

const isIOS = Utils.isIOS;
export const styles = StyleSheet.create({
    container: {
        paddingLeft: isIOS ? 5 : 20,
        paddingRight: isIOS ? 5 : 20,
        transform: I18nManager.isRTL ? [{ rotate: '180deg' }] : [],
        width: 25,
        height: 25,
        marginLeft: 20,
        alignItems: 'center',
        justifyContent: 'center',
    },
});