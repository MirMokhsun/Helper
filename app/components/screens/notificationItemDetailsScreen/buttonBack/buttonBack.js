import React, { Component } from 'react';
import { View, StyleSheet, TouchableOpacity, I18nManager } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PreviousAndroid from './previousAndroid';
import PreviousIOS from './previousIOS';
import { withNavigation } from 'react-navigation';
import Utils from '../../../../api/utils/utils';

const { isIOS } = Utils;

const styles = StyleSheet.create({
    container: {
        width: 56,
        height: '100%',
        justifyContent: 'center',
        paddingLeft: isIOS ? 5 : 20,
        paddingRight: isIOS ? 5 : 20,
        transform: I18nManager.isRTL ? [{ rotate: '180deg' }] : [],
    },
});

export class ButtonBack extends Component {
    constructor(props) {
        super(props);
        this.previousSvg = this.getPreviousSvg();
    }

    goBack = () => {
        const { navigation, userEmail } = this.props;
        const routNames = navigation.dangerouslyGetParent().state.routes;
        const currentRoutName = routNames[routNames.length - 1].routeName;
        if (currentRoutName === 'NotificationsItemDetailsScreen') {
            navigation.navigate('NotificationsScreen');
        } else if (userEmail) {
            navigation.navigate('TabAppNavigator');
        } else {
            navigation.navigate('RegistrationScreen');
        }
    }

    getPreviousSvg = () => {
        let result = <PreviousAndroid width={16} height={16} />;
        if (isIOS) {
            result = <PreviousIOS width={20} height={20} />;
        }
        return result;
    }

    render() {
        const { id } = this.props;
        return (
            <TouchableOpacity onPress={this.goBack} testID={`NotificationButtonBack${id}ID`} accessibilityLabel={`NotificationButtonBack${id}ID`} >
                <View style={styles.container}>
                    {this.previousSvg}
                </View>
            </TouchableOpacity >
        );
    }
}

const mapStateToProps = state => ({
    historyNotification: Object.values(state.historyNotification),
    notificationObj: state.setNotificationObj,
    userEmail: state.userLoginData.userEmail,
});
const mapDispatchToProps = dispatch => bindActionCreators({
}, dispatch);
export default withNavigation(connect(mapStateToProps, mapDispatchToProps)(ButtonBack));