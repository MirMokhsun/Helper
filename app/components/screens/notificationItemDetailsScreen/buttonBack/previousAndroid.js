import React from "react";
import Svg, { Path } from "react-native-svg";

const PreviousAndroid = ({ width, height }) => (
<Svg width={ width || 408} height={ height || 408} viewBox="0 0 408 408"><Path d="M408 178.5H96.9L239.7 35.7 204 0 0 204l204 204 35.7-35.7L96.9 229.5H408v-51z" fill="#000756"/></Svg>
);

export default PreviousAndroid;