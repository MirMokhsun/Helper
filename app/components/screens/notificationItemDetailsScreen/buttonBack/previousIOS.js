import React from "react";
import Svg, { Path } from "react-native-svg";

const PreviousIOS  = ({ width, height }) => (
<Svg width={ width || 55.753} height={ height || 55.753} viewBox="0 0 55.753 55.753"><Path d="M12.745 23.915c.283-.282.59-.52.913-.727L35.266 1.581a5.4 5.4 0 1 1 7.637 7.638L24.294 27.828l18.705 18.706a5.4 5.4 0 0 1-7.636 7.637L13.658 32.464a5.367 5.367 0 0 1-.913-.727 5.367 5.367 0 0 1-1.572-3.911 5.369 5.369 0 0 1 1.572-3.911z" data-original="#000000" class="active-path" data-old_color="#F9ECEC" fill="#000756"/></Svg>
);

export default PreviousIOS;