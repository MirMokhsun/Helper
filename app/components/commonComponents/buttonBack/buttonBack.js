import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { styles } from './styles';
import { withNavigation } from 'react-navigation';
import Utils from '../../../api/utils/utils';
import ArrowBack from '../svg/ArrowBack';
import ArrowBackIOS from '../svg/ArrowBackIOS';

const { isIOS } = Utils;

export class ButtonBack extends React.Component {
    constructor(props) {
        super(props);
        const { color } = props;
        this.previousSvg = this.getPreviousSvg( color );
    }

    goBack = () => {
        const { navigation } = this.props;
        navigation.goBack();
    }

    getPreviousSvg = (color) => {
        let result = <ArrowBack  color = {color || '#fff'}  />;
        if (isIOS) {
            result = <ArrowBackIOS color = {color || '#fff'} width={20} height={20} />;
        }
        return result;
    }

    render() {
        const { id } = this.props;
        return (
            <TouchableOpacity testID = {`ButtonBack${id}ID`} accessibilityLabel={`ButtonBack${id}ID`} onPress={this.goBack}>
                <View style={styles.container}>
                    {this.previousSvg}
                </View>
            </TouchableOpacity >
        );
    }
}

export default withNavigation(ButtonBack);