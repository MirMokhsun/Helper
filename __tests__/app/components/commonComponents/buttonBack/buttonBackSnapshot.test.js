import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ButtonBack } from '../../../../../app/components/commonComponents/buttonBack/buttonBack';

it('create snapshot ButtonBack', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<ButtonBack navigation={() => { }} />);
    expect(toolbarSnapshot).toMatchSnapshot();
});