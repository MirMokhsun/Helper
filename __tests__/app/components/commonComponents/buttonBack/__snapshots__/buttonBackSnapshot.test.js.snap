// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot ButtonBack 1`] = `
<TouchableOpacity
  accessibilityLabel="ButtonBackundefinedID"
  activeOpacity={0.2}
  onPress={[Function]}
  testID="ButtonBackundefinedID"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 25,
        "justifyContent": "center",
        "marginLeft": 20,
        "paddingLeft": 5,
        "paddingRight": 5,
        "transform": Array [],
        "width": 25,
      }
    }
  >
    <ArrowBackIOS
      color="#fff"
      height={20}
      width={20}
    />
  </View>
</TouchableOpacity>
`;
